src/fr.rs

```diff
@@ -398,7 +398,12 @@ impl Fr {
         rand.fill_bytes(&mut bytes);
         // Ensure that the value is lower than `L`.
         bytes[31] &= 0b0000_0001;
-        Fr::from_bytes(&bytes).unwrap_or_else(|| Fr::random(rand))
+
+        Fr::from_bytes(&bytes).unwrap_or({
+            let mut bytes = [0u8; 64];
```

Comment:
This implementation always created both arrays (one of 32 bytes and the second of 64 bytes) executing `rand.fill_bytes` for both of them.
I think it's a waste, at this point we could just reduce the code to just the implementation we know won't fail:

```rust
pub fn random<T>(rand: &mut T) -> Fr
where
    T: RngCore + CryptoRng,
{
    let mut bytes = [0u8; 64];
    rand.fill_bytes(&mut bytes);
    Fr::from_bytes_wide(&bytes)
}
```